baker/SiteBaker.tsx

```diff
@@ -620,59 +620,83 @@ export class SiteBaker {
                         IMAGE_HOSTING_BUCKET_SUBFOLDER_PATH,
                         image.filename
                     )
-                    return retryPromise(() => fetch(remoteFilePath))
-                        .then((response) => {
-                            if (!response.ok) {
-                                throw new Error(
-                                    `Fetching image failed: ${response.status} ${response.statusText} ${response.url}`
-                                )
-                            }
-                            return response.arrayBuffer()
+                    const localImagePath = path.join(
+                        imagesDirectory,
+                        image.filename
+                    )
+                    const localImageEtagPath = localImagePath + ".etag"
+
+                    // If the image already exists locally, try to use its etag
+                    const existingEtag =
+                        fs.existsSync(localImagePath) &&
+                        fs.existsSync(localImageEtagPath)
+                            ? await fs.readFile(localImageEtagPath, "utf8")
+                            : ""
```

Comment:
Not that it makes a massive difference here, but I think we should use async methods whenever possible, not least because it will make it a bit easier to parallelize things in the future.
```suggestion
                    const existingEtag = await Promise.all([
                        fs.exists(localImagePath),
                        fs.exists(localImageEtagPath),
                    ]).then(([exists, etagExists]) =>
                        exists && etagExists
                            ? fs.readFile(localImageEtagPath, "utf8")
                            : ""
                    )
```

Reply:
Thanks, looks better!